GVFS/GVFS/RepairJobs/GitRefsRepairJob.cs

```diff
@@ -0,0 +1,174 @@
+﻿using System;
+using GVFS.Common;
+using GVFS.Common.Git;
+using GVFS.Common.Tracing;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace GVFS.RepairJobs
+{
+    public abstract class GitRefsRepairJob : RepairJob
+    {
+        protected GitRefsRepairJob(ITracer tracer, TextWriter output, GVFSEnlistment enlistment) 
+            : base(tracer, output, enlistment)
+        {
+        }
+
+        public override IssueType HasIssue(List<string> messages)
+        {
+            int numBadRefs = 0;
+
+            foreach (string @ref in this.GetRefs())
+            {
+                if (!this.TryParseRef(@ref, messages))
+                {
+                    numBadRefs++;
+                }
+            }
+
+            return numBadRefs > 0 ? IssueType.Fixable : IssueType.None;
+        }
+
+        /// <summary>
+        /// Fixes the refs using the reflog to find the last SHA.
+        /// </summary>
+        public override FixResult TryFixIssues(List<string> messages)
+        {
+            int numFailures = 0;
+
+            foreach (string @ref in this.GetRefs())
+            {
+                // We should only attempt to fix bad refs
+                if (!this.TryParseRef(@ref))
+                {
+                    if (!this.TryWriteRefFromLog(@ref, messages))
+                    {
+                        numFailures++;
+                    }
+                }
+            }
+
+            if (numFailures > 0)
+            {
+                messages.Add($"Not all references could be fixed. Failed to fix {numFailures} references.");
+                return FixResult.Failure;
+            }
+
+            return FixResult.Success;
+        }
+
+        /// <summary>
+        /// Get the list of full symbolic references (eg. HEAD, refs/remotes/origin/HEAD, refs/heads/master)

```

Comment:
The example is somewhat confusing since none of the code inspects the `refs\remotes` folder.

Reply:
Even if it was pre-existing, I would drop the examples. It's potentially quite misleading to a future reader who may think that it would be desirable to repair remote refs, when it definitely would not be.